artman/config/loader.py

```diff
@@ -56,12 +64,19 @@ def _read_artman_config(artman_yaml_path):
 
 def _parse(artman_yaml_path):
     """Parse artman yaml config into corresponding protobuf."""
-    with open(artman_yaml_path, 'r') as f:
-        # Convert yaml into json file as protobuf python load support paring of
-        # protobuf in json or text format, not yaml.
-        artman_config_json_string = json.dumps(yaml.load(f))
-    config_pb = Config()
-    json_format.Parse(artman_config_json_string, config_pb)
+    if not os.path.exists(artman_yaml_path):
+      raise ValueError(CONFIG_NOT_FOUND_ERROR_MESSAGE_FORMAT % artman_yaml_path)
+
+    try:
+        with open(artman_yaml_path, 'r') as f:
+            # Convert yaml into json file as protobuf python load support paring of
+            # protobuf in json or text format, not yaml.
+            artman_config_json_string = json.dumps(yaml.load(f))
+        config_pb = Config()
+        json_format.Parse(artman_config_json_string, config_pb)
+    except (json_format.ParseError, yaml.parser.ParserError) as e:
+        logger.error(INVALID_CONFIG_ERROR_MESSAGE_FORMAT % artman_yaml_path)
+        raise ValueError(str(e))
```

Comment:
Why not just `raise`?

Reply:
If the artman yaml input goes wrong, it is most likely cased by json_format.ParseError. Then it could cause confusing to raise a json format error to the caller while the input is a yaml file.

OTOH, I would agree that ValueError is not good either. If artman is going to be used as a library not a CLI, I would consider create some custom error type.

Reply:
I am more concerned that you are discarding both the stack trace and any additional attributes on the original exception when you create a new error and stringify the old error. That can potentially make debugging much more difficult.

Reply:
I would say the stack trace in this particular case is not quite helpful to the artman users. Below is an example error output under current implementation.

$artman --config google/cloud/speech/v1/cloud_speech_gapic.yaml generate java_gapic
artman> Artman YAML google/cloud/speech/v1/cloud_speech_gapic.yaml is invalid.
artman> Artifact config loading failed with `Message type "googleapis.artman.Config" has no field named "interfaces".
 Available Fields(except extensions): <MessageFields sequence>`

I agree that the stack trace would be useful for some cases. I logged the trace information at debug logging level. When one pass --verbose, they will see the full stacktrace.

Reply:
PTAL, @geigerj

Reply:
I still don't understand the advantage of throwing away the original error here. If anything else goes wrong, it's useful to have the stacktrace, and it's still seems to be me this is throwing away a potentially rich exception object and replace it with its string representation for no real benefit.

Reply:
What is the value of printing the stacktrace to sys.err if I am a just regular artman user? Doesn't `Message type "googleapis.artman.Config" has no field named "interfaces"` already tell everything artman could return? Or I might misread your comment. Do you want such exceptions be captured and handled by artman, you would suggest just raise it so that artman execution will exit with a full stacktrace?

Reply:
> What is the value of printing the stacktrace to sys.err if I am a just regular artman user? 

It lets you set breakpoints if you're using a debugger, for one. These exceptions are also objects, so you can catch and inspect them normally, but once you throw away the original object, there's no way to retrieve any attributes that might not be captured in its string representation.

FWIW, this is solved in Python 3 with the `raise ... from ...` syntax.